### PR TITLE
iOS7 Add pointer device support

### PR DESCRIPTION
**Pull request by larsamannen**

### CODE REVIEW

File: backends/platform/ios7/ios7_game_controller.mm
Comment (sev-) [inline backends/platform/ios7/ios7_game_controller.mm:76]: A little nitpick regarding the code formatting, we do not do an extra indent for the `switch()` labels

**Comment by ccawley2011** [comment]
For game controllers, I'd recommend sending controller events to allow the iOS port to make better use of the keymapper instead of using a hard-coded mapping in the iOS 7 backend. There are some details about this here: https://wiki.scummvm.org/index.php?title=Keymapper

File: backends/platform/ios7/ios7_game_controller.mm
Comment (larsamannen) [reply backends/platform/ios7/ios7_game_controller.mm:76]: Will fix

**Comment by larsamannen** [comment]
By the way, I forgot to mention that I’ve tested the implementation on an iPad Pro, using the magic keyboard, a Magic Mouse and a Nimbus controller.

File: devtools/create_project/xcode.cpp
Comment (criezy) [inline devtools/create_project/xcode.cpp:559]: `configure` and `ports.mk` also need to be updated for those not using Xcode to build the package (for example on our buildbot).

**Review by criezy — COMMENTED** [review]
I had a very quick look and will look more in details in the next few days.
Overall the code quality looks good.

I already added a few comments though, and it also looks like you intended to squash 94d0e74d397bd509209b9544b7af122af3fd5fb6 but forgot?

Note that I will not be able to test the code as my iPad is too old and stuck to iOS 12. But I can at least check that it compiles and still works.

File: backends/platform/ios7/ios7_video.mm
Comment (criezy) [reply backends/platform/ios7/ios7_video.mm:427]: I forgot to indicate here that after my compilation tests from today, this was not an issue for me. So I am guessing this would only be an issue for very old Xcode versions.

**Comment by larsamannen** [comment]
Hmmmm, the gamepad controller support should not be compiled on iOS versions older than 14 due to the API-check in the header file. But I will look at it. 

I will also check the regression problem with touchpad mode disabled as well. 

I’m currently on a trip so I will look into it late next week. 
Thanks for the review

File: backends/platform/ios7/ios7_video.mm
Comment (criezy) [reply backends/platform/ios7/ios7_video.mm:427]: We have a history of supporting old platforms. However that happens when we have a dedicated developer spending time to keep ScummVM working on those. And even then they might require a modern compiler (we at least require support for c++11).

For iOS we don't even have a maintainer for modern iOS and Xcode versions, let alone old ones. So I don't have an issue with dropping support for using very old Xcode version that do not support the `@available` keyword (I suspect that would mean any version older than Xcode 9, which is probably the first version to support it since it was released in 2017).

**Comment by sev-** [comment]
@larsamannen hope you will be able to look into it soon

**Comment by larsamannen** [comment]
> @larsamannen hope you will be able to look into it soon

I’m currently on a vacation trip. Will be able to look into it end of next week if that’s ok?

**Comment by criezy** [comment]
Thank you for the updates. It looks good to me now. And I also checked that it now compiles with Xcode 9.2. I just want to play a bit with it on my iPad before I approve this pull request.

**Review by criezy — APPROVED** [review]
Thanks a lot for your work.
I can confirm it behaves very nicely on my iPad now.
I have not tested the changes in `configure` and `ports.mk` but they look good and I will be monitoring buildbot after the merge to check that they work as expected.

So this looks all good to me now, and I will be merging the pull request.

Edit: I will be waiting for the checks, and in particular the iOS build, to complete before merging.

**Pull request merged**
